src/client/transport.js

```diff
@@ -166,21 +166,21 @@ class Transport extends EventEmitter {
         sendMsg();
     }
 
-    waitCookieMsg () {
-        return new Promise(resolve => {
-            var handler = () => {
-                if (!this._cookieMsgInProgress()) {
-                    this.off(this.MSG_RECEIVED_EVENT, handler);
-
-                    resolve();
-                }
-            };
+    // TODO: Don't use the Promise because if the queue of cookie message is empty,
+    // we need call a callback function synchronously (GH-722)
```

Comment:
So, you got rid of the promise. Then it's maybe a `NOTE` not `TODO`?

```
    // NOTE: We don't use a promise here because if the cookie message queue is empty,
    // we need to call a callback function synchronously (GH-722)
```